lib/liquid/filters.ex

```diff
@@ -80,4 +448,24 @@ defmodule Liquid.Filters do
     filter(rest, ret)
   end
 
+  def filter(a, b, _) when a == [] do
+    filter(a, b)
+  end
+
+  def filter([head|tail], b, context) do
+    case context.assigns do
```

Comment:
I think you really should use Liquid.Variable for parsing the arguments instead of custom code here to parse the arguments. The Liquid.Variable create and lookup will correctly handle the value parsing like any other variable anywhere else in liquid. 

It makes sure that everything is parsed the same and we can optimize it at 1 location if it needs it to be.